pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,533 @@
+				(&account_id, &currency_id, &receiver),
+				&allowance_details,
+			);
+
+			Self::deposit_event(Event::TransferAllowanceCreated {
+				sender_account_id: account_id,
+				currency_id,
+				receiver,
+				allowed_at: allowance_details.allowed_at,
+				blocked_at: allowance_details.blocked_at,
+			});
+			Ok(())
+		}
+
+		/// Restricts a transfer allowance for a sending account/currency/receiver location to:
+		/// - either the current block + delay if a delay is set
+		/// - or the current block if no delay is set
+		#[pallet::call_index(1)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 2).ref_time())]
+		pub fn remove_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			receiver: T::Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			let blocked_at = match Self::get_account_currency_restriction_count_delay(
+				&account_id,
+				currency_id,
+			) {
+				Some((_, Some(delay))) => {
+					<frame_system::Pallet<T>>::block_number().saturating_add(delay)
+				}
+				_ => <frame_system::Pallet<T>>::block_number(),
+			};
+			match <AccountCurrencyTransferAllowance<T>>::get((&account_id, &currency_id, &receiver))
+			{
+				Some(existing_allowance) => {
+					let allowance_details = AllowanceDetails {
+						blocked_at,
+						..existing_allowance
+					};
+					<AccountCurrencyTransferAllowance<T>>::insert(
+						(&account_id, &currency_id, &receiver),
+						&allowance_details,
+					);
+					Self::deposit_event(Event::TransferAllowanceRemoved {
+						sender_account_id: account_id,
+						currency_id,
+						receiver,
+						allowed_at: allowance_details.allowed_at,
+						blocked_at: allowance_details.blocked_at,
+					});
+					Ok(())
+				}
+				None => Err(DispatchError::from(Error::<T>::NoMatchingAllowance)),
+			}
+		}
+
+		/// Removes a transfer allowance for a sending account/currency and receiving location
+		/// Decrements or removes the sending account/currency count.
+		#[pallet::call_index(2)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1, 2).ref_time())]
+		pub fn purge_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			receiver: T::Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			match <AccountCurrencyTransferAllowance<T>>::get((&account_id, &currency_id, &receiver))
+			{
+				Some(_) => {
+					T::ReserveCurrency::unreserve(
+						&account_id,
+						T::Fees::fee_value(T::AllowanceFeeKey::get()),
+					);
+					<AccountCurrencyTransferAllowance<T>>::remove((
+						&account_id,
+						&currency_id,
+						&receiver,
+					));
+					Self::decrement_or_remove_allowance_count(&account_id, &currency_id)?;
+					Self::deposit_event(Event::TransferAllowancePurged {
+						sender_account_id: account_id,
+						currency_id,
+						receiver,
+					});
+					Ok(())
+				}
+				None => Err(DispatchError::from(Error::<T>::NoMatchingAllowance)),
+			}
```

Comment:
This should ONLY be possible of the `blocked_at` has already passed!

Reply:
👍 updated to ensure `blocked_at` has already passed.